tensorflow/core/lib/strings/str_util.cc

```diff
@@ -416,6 +417,118 @@ bool ConsumeNonWhitespace(StringPiece* s, StringPiece* val) {
   }
 }
 
+Status SplitUTF8(StringPiece text, const string& delim,
+                 std::vector<string>* result) {
+  // Bytes    Byte 1    Byte 2    Byte 3    Byte 4
+  //   1     0xxxxxxx
+  //   2     110xxxxx  10xxxxxx
+  //   3     1110xxxx  10xxxxxx  10xxxxxx
+  //   4     11110xxx  10xxxxxx  10xxxxxx  10xxxxxx
+  result->clear();
+  result->reserve(text.size());
+  size_t off = 0, len = 0;
+  bool advance = true;
+  for (size_t i = 0; i < text.size(); ++i) {
+    if (off == i) {
+      size_t num_bytes = 0;
+      Status s = UTF8CharNumBytes(text.substr(off, 4), &num_bytes);
```

Comment:
Why `text.substr(off, 4)` here?
That might overflow `text` no? And we could just as well provide `text` (i.e, without `substr`)?